src/System.Text.Json/src/System/Text/Json/JsonCommentHandling.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text.Json
+{
+    /// <summary>
+    /// This enum defines the various ways the <see cref="JsonUtf8Reader"/> can deal with comments.
+    /// </summary>
+    public enum JsonCommentHandling : byte
+    {
+        /// <summary>
+        /// By default, do no allow comments within the JSON input.
+        /// Comments are treated as invalid JSON if found and a
+        /// <see cref="JsonReaderException"/> is thrown.
+        /// </summary>
+        Default = 0,
```

Comment:
Since the enum is named `JsonCommentHandling`, should we have shorter value names? e.g.
```C#
public enum JsonCommentHandling
{
    /// <summary>Comments within the JSON input are treated as invalid input.  This is the default.</summary>
    Invalid = 0,
    /// <summary>Comments within the JSON input are treated as valid tokens.</summary>
    Read = 1,
    /// <summary>Comments within the JSON input are ignored.</summary>
    Skip = 2
}
```
?

Just throwing it out there... I'm fine with the current names, too.

Reply:
I don't like `Invalid` as a default name for it. I also prefer `Allow` over Read. But we can certainly remove the `Comments` suffixes.

Reply:
I like `Disallow`